### PR TITLE
Feature/extract mobile page layout

### PR DESCRIPTION
**Pull request by Toflar**
I've extracted the mobile page layout feature into its own bundle: https://github.com/Toflar/contao-mobile-page-layout-bundle

To do so, I had to implement two things:

* Ability for third party bundles to add a callback that allows to configure inheritance during `PageModel::loadDetails()`. I could've used this several times before anyway.
* The `{{toggle_view}}` insert tag must never be cached but so far we had no possibility in the current insert tag handling to make sure one can make their own insert tag uncacheable without having to set the `|uncached` flag. I've invented an additional insert tag named `{{esi::<real-inserttag>}}` which allows to wrap your insert tag. That way, you can force Contao to re-use the ESI request when the `$blnCache` flag is set accordingly. See https://github.com/Toflar/contao-mobile-page-layout-bundle/blob/master/src/EventListener/InsertTags/ToggleViewListener.php#L58 for an example.

### CODE REVIEW

File: core-bundle/src/Resources/contao/dca/tl_article.php [hunk 1]
```diff
@@ -627,25 +627,20 @@ public function getActiveLayoutSections(Contao\DataContainer $dc)
...
-					continue;
-				}
-
-				$objLayout = Contao\LayoutModel::findByPk($objPage->$key);
+				$objLayout = Contao\LayoutModel::findByPk($objPage->layout);
 
 				if ($objLayout === null)
 				{
-					continue;
+					return array();
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/dca/tl_article.php:636]: Are you sure that this is correct?

File: core-bundle/src/Resources/contao/models/PageModel.php [hunk 2]
```diff
@@ -919,24 +912,20 @@ public function loadDetails()
 						$this->clientCache = $this->clientCache !== false ? $this->clientCache : $objParentPage->clientCache;
 					}
 
+					// Protection
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/models/PageModel.php:922]: Why did you move this code block?

File: core-bundle/src/Resources/contao/pages/PageRegular.php [hunk 3]
```diff
@@ -239,31 +229,27 @@ protected function prepare($objPage)
...
+		$objLayout = LayoutModel::findByPk($objPage->layout);
 
-		// Override the autodetected value
-		if (Input::cookie('TL_VIEW') == 'mobile')
-		{
-			$blnMobile = true;
-		}
-		elseif (Input::cookie('TL_VIEW') == 'desktop')
+		// HOOK: modify the page or layout object (see #4736)
+		if (isset($GLOBALS['TL_HOOKS']['getPageLayout']) && \is_array($GLOBALS['TL_HOOKS']['getPageLayout']))
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/pages/PageRegular.php]: Why did you move the hook?

File: core-bundle/src/Resources/contao/dca/tl_article.php (hunk 1 above)
Comment (Toflar) [reply core-bundle/src/Resources/contao/dca/tl_article.php:636]: Yes, it's not a loop anymore so if there's nothing to load the options from, it's going to be an empty array.

File: core-bundle/src/Resources/contao/models/PageModel.php (hunk 2 above)
Comment (Toflar) [reply core-bundle/src/Resources/contao/models/PageModel.php:922]: Guess that must have been an issue during rebase. Fixed in 92007bc2231ed6fe6621812ce1130c394b347895.

File: core-bundle/src/Resources/contao/pages/PageRegular.php (hunk 3 above)
Comment (Toflar) [reply core-bundle/src/Resources/contao/pages/PageRegular.php]: Because the method says `getPageLayout()`. So I figured that's the correct way to place the hook. The order is still the same, it finds the assigned layout and then calls the hook.

**Comment by leofeyer** [comment]
Thank you @Toflar.
